packages/typespec-azure-core/src/rules/no-generic-types.ts

```diff
@@ -0,0 +1,38 @@
+import { Model, createRule, paramMessage } from "@typespec/compiler";
+
+const disallowList = new Set(["integer", "numeric", "float", "decimal"]);
+const alternatives = new Map([
+  ["integer", "int32"],
+  ["numeric", "int32"],
+  ["float", "float32"],
+  ["decimal", "float32"],
+]);
+
+export const noGenericTypesRule = createRule({
+  name: "no-generic-types",
+  description: "Don't use generic types. Use more specific types instead.",
+  severity: "warning",
+  url: "https://azure.github.io/typespec-azure/docs/libraries/azure-core/rules/no-generic-types",
+  messages: {
+    default: paramMessage`Don't use generic type '${"name"}'. Use a more specific type that specifies the bit size, such as '${"alternative"}' instead.`,
+  },
+  create(context) {
+    return {
+      model: (model: Model) => {
+        for (const [_, prop] of model.properties) {
+          if (prop.type.kind === "Scalar") {
+            if (disallowList.has(prop.type.name)) {
+              context.reportDiagnostic({
+                target: prop.type,
+                format: {
+                  name: prop.type.name,
+                  alternative: alternatives.get(prop.type.name)!,
```

Comment:
I think with the alternative we could provide the codefix. shopuld be easy we already provide the codefix from changing an identifier `createChangeIdentifierCodeFix` (used today to change `number` to `float64`)

```ts
return [createChangeIdentifierCodeFix(node, "float64")];
```

Reply:
I can certainly do that, but do we really want to put _something_ there just to resolve the rule if it's potentially wrong? Would it be better to just force the service team to do the research and put the size-appropriate value?

Reply:
I think we need the service team to specify the actual behavior. How many instances of existing specs are running into this issue?